### PR TITLE
fix(ci): keep the scorecard job within the allowlist scorecard.dev publishes from

### LINKED ISSUE #514: bug: the OpenSSF Scorecard score has not been published since 2026-08-06
### Pre-flight checklist

- [x] I searched existing issues and did not find a duplicate.
- [x] This is a bug, not a security vulnerability or a usage question.

### What happened?

The README's OpenSSF Scorecard badge and https://scorecard.dev/viewer/?uri=github.com/zuke-build/zuke show a score of 6.6 computed on 2026-08-06 at commit 9b213be, and it never moves. The `Scorecard supply-chain` workflow runs green on every push to master and on its weekly schedule, but every run since then has failed to publish. Today's run (https://github.com/zuke-build/zuke/actions/runs/34226583571) computed a score of 8.7 and then logged:

```
error sending scorecard results to webapp: http response 400, status: 400 Bad Request,
error: {"code":400,"message":"workflow verification failed: job has unallowed step: zuke-build/zuke,
see https://github.com/ossf/scorecard-action#workflow-restrictions for details."}
::warning::Unable to POST scorecard results to webapp ...
```

The action only emits a warning on a rejected publish, so the job succeeds and nothing surfaced it.

scorecard.dev verifies the workflow server-side before accepting a result. The job that contains `ossf/scorecard-action` may only have `uses:` steps, and only from an allowlist: `actions/checkout`, `actions/create-github-app-token`, `actions/upload-artifact`, `github/codeql-action/upload-sarif`, `step-security/harden-runner`, `ossf/scorecard-action`. No `run:` steps and no job-level `env`. The current `scorecard.yml` violates this twice:

- #302 (2026-08-08) replaced harden-runner and checkout with the `zuke-build/zuke` composite action, which is the step the error names.
- #295 (2026-08-06) replaced `github/codeql-action/upload-sarif` with a `run: ./zuke scorecardSarif` step carrying a step-level `GITHUB_TOKEN`. The verifier rejects any `run:` step in that job.

The last accepted publish is the commit immediately before #295 landed.

### Expected behaviour

Every push to master and every scheduled run publishes the score, so the badge and the viewer track the current state of the repository.

### Minimal reproduction

```typescript
// Push any commit to master, then open the "Run Scorecard" step of the
// Scorecard supply-chain workflow run and search its log for
// "Unable to POST scorecard results to webapp".
```

### Affected package(s)

None published. The repository's own generated workflow, declared in `build/workflows.ts` (`scorecard`), and the `scorecardSarif` target in `zuke.ts`.

### Proposed fix

Bring the scorecard job back within the allowlist without hand-writing YAML: opt that one job out of the Zuke prelude by naming `step-security/harden-runner` and `actions/checkout` explicitly, and replace the target's run step with the `ossf/scorecard-action` and `github/codeql-action/upload-sarif` pair via the job's `steps`, the same shape the `codeql` job already uses. The `scorecardSarif` target loses its CI caller and is retired. Add a regression test that asserts the committed `scorecard.yml` job contains only allowlisted `uses:` steps, no `run:` step, and no job-level `env`, so the publish cannot be broken silently again.

### PR DESCRIPTION
**Pull request by totollygeek**
## What & why

The README's OpenSSF Scorecard badge has shown 6.6, computed on 2026-08-06, for a month while the `Scorecard supply-chain` workflow ran green on every push. Today's run computed 8.7 and then logged, as a warning only, that it was unable to POST the results to the webapp: HTTP 400, `workflow verification failed: job has unallowed step: zuke-build/zuke`.

scorecard.dev verifies the workflow server-side before it accepts a published score. The job holding `ossf/scorecard-action` may consist only of `uses:` steps from a short allowlist: `actions/checkout`, `actions/create-github-app-token`, `actions/upload-artifact`, `github/codeql-action/upload-sarif`, `step-security/harden-runner`, and the scorecard action itself, with no `run:` step and no job-level `env`. Two changes had pushed the job outside it: #302 replaced harden-runner and checkout with the `zuke-build/zuke` composite action, and #295 replaced `upload-sarif` with a `./zuke scorecardSarif` run step. Every publish since then was refused.

This brings the job back within the allowlist without hand-writing YAML:

- The `scorecard` declaration in `build/workflows.ts` opts that one job out of the Zuke prelude by naming `step-security/harden-runner` and `actions/checkout` explicitly, which the generator renders as separate pinned steps, and uses `steps` to replace the target's run step with the `ossf/scorecard-action` and `github/codeql-action/upload-sarif` pair. This is the same shape the `codeql` job already uses. `scorecard.yml` is regenerated and matches the last workflow scorecard.dev accepted, minus the optional `upload-artifact` copy of the SARIF, which the Security tab upload covers.
- The `scorecardSarif` target is retired. Its CI caller is gone, so it becomes a `scorecard` target that, like `codeql`, fails locally with an explanation rather than pretending to run. The `uploadSarif` task in `@zuke/gh` is unaffected.
- `github/codeql-action/upload-sarif` joins `SEED_PINS`, at the release the committed `codeql.yml` already pins; the `init` and `analyze` seeds move to the same SHA so the three subpaths of one action agree. Dependabot keeps owning the pin from the generated file onward.
- `docs/graph.md` regenerated.

Verified locally: `format`, `lint`, `spell`, `coverage`, `apiDocsCheck`, `docLint`, `snippetsCheck`, `examplesCheck`, `hclSyncCheck`, `pluginSyncCheck`, `skillsCheck`, `graphDocCheck`, `prBodyLint`, `actionPinCheck`, and `lockCheck` all pass. The `security` target could not complete in the sandbox because zizmor's advisory lookup against the GitHub API is blocked there; zizmor run offline with the gate's persona and actionlint both report the regenerated `scorecard.yml` clean. The workflow's `Scorecard supply-chain` run on the merge commit is the real proof: its `Run Scorecard` step should no longer log `Unable to POST scorecard results to webapp`, and the badge should move to the current score on the first push to master after merge.

### Tests

- `tests/scorecard_workflow_test.ts` holds the committed `scorecard.yml` to the allowlist: only permitted actions and no `run:` step, no job-level `env`, `container` or `services`, a hosted Ubuntu runner, `id-token: write` on the scorecard job alone with no workflow-level write permission, and `publish_results` set with the upload reading the file the scorecard step wrote. It finds the job by its step rather than its id, the way the verifier does.
- `tests/_workflow.ts` is the shared reader for committed workflows, parsing a workflow, its jobs, their steps and the action a step uses, so this test and `tests/action_release_job_test.ts` share one implementation instead of each retyping it.
- `tests/action_pins_test.ts` gains a test that the `github/codeql-action/*` seeds pin one SHA.

## Related issues

Closes #514

## Checklist

- [x] This PR closes a tracking issue (`Closes #<n>` above).
- [x] The PR title is a
      [Conventional Commit](https://www.conventionalcommits.org/)
      (`type(scope): summary`).
- [x] `deno task ci` passes locally (lint, fmt, type-check, tests, spell). Every target except `security`, whose GitHub advisory lookup is blocked in the sandbox; see above.
- [x] Tests were added or updated; coverage stays at 95%+ (lines and branches).
- [x] Docs updated in the same PR (`README.md`, JSDoc, `docs/`) when behaviour
      changed.
- [x] Public API changes were regenerated with `./zuke apiDocs` (`llms.txt`,
      `llms-full.txt`, package README `## API`). Not applicable: no package API changed.
- [x] No `any`, no `as` casts or `!` non-null assertions in `src/` (narrow with
      type guards instead).
- [x] No dead code: unreachable branches and can't-fire fallbacks are removed,
      with types tightened so the impossible state is unrepresentable.
- [x] No duplicated logic: an existing helper is imported rather than retyped,
      and a near-copy differing by a constant or a message is parameterised into
      one implementation. A guard, escape, or credential resolution has exactly
      one implementation.
- [x] SOLID shapes respected: one domain per file, extension via the settings
      lambda (not a behaviour-switching flag), narrow parameter types, and
      dependencies on the injectable seam (`StateHost`, an injected `fetch`,
      `EnvReader`) rather than `Deno.*` or a global.
- [x] A new package was wired into all six places (see
      [AGENTS.md](../blob/master/AGENTS.md#good-open-source-practices-to-follow)),
      if applicable. Not applicable.
- [x] The code is written using AI assisted coding.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_012tpd9Bj4X2gMXNbzJstvtN

### CODE REVIEW

File: tests/scorecard_workflow_test.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+/**
+ * The only actions scorecard.dev permits in the publishing job — the action
+ * itself, the two prelude halves, and the two ways of getting its report out.
+ */
+const ALLOWED_ACTIONS: ReadonlySet<string> = new Set([
+  SCORECARD_ACTION,
+  "actions/checkout",
+  "actions/create-github-app-token",
+  "actions/upload-artifact",
+  "github/codeql-action/upload-sarif",
```
[the remark this answers is not in the document]
Comment (totollygeek) [reply tests/scorecard_workflow_test.ts:49]: Fixed `2h53gaox7b2he` in the follow-up commit. The YAML reading (workflow, jobs, steps, the action a step uses) now lives once in `tests/_workflow.ts`, imported by both this test and `tests/action_release_job_test.ts`, which was refactored onto it. The full-SHA pin assertion is dropped from this file: the security gate's zizmor unpinned-uses audit already enforces it for every committed workflow. And a step with no `uses` now reads as a `run:` step rather than as an empty action name, so that edge case no longer differs between the two tests.

---
_Generated by [Claude Code](https://claude.ai/code)_
